apps/theming-designer/src/components/SemanticSlots.tsx

```diff
@@ -432,33 +44,116 @@ export const SemanticSlots: React.StatelessComponent<ISemanticSlotsProps> = (pro
     );
   };
 
-  const fillVariantSlotsList = (variantType: VariantThemeType, mappingType: {}): JSX.Element[] => {
+  const trimPaletteColors = (paletteColors: {}): {} => {
+    let trimmedPaletteColors = {};
+    for (let palette in paletteColors) {
+      if (
+        palette.startsWith('theme') ||
+        palette.startsWith('neutral') ||
+        palette === 'black' ||
+        palette === 'white' ||
+        palette === 'redDark' ||
+        palette === 'accent'
+      ) {
+        (trimmedPaletteColors as any)[palette] = (paletteColors as any)[palette];
+      }
+    }
+    return trimmedPaletteColors;
+  };
+
+  const trimSemanticColors = (semanticColors: {}): {} => {
```

Comment:
I combined them by making trimSemantic take in a {} | string[] and return {} | string[] since that's what I need for it to work for both making the final list of slot names as well as the mapping of semantic slots to their values. But it throws an error saying Type '{} | string[]' is not assignable to type 'string[]' so I had to make (slotNames as any) to resolve that... not sure if this is the best trade off since earlier comment says to remove all anys.

Reply:
things get confusing when you return a union type, that means the code that takes the return value needs to be able to handle both, because you're potentially getting either one
you'll want to use generics in order to say "if I get type T1, I'll return type T1 as well"

Reply:
done in polished PR https://github.com/OfficeDev/office-ui-fabric-react/pull/10888/files#diff-8565a4a2faf78e1078e6f323d0438f9eR97